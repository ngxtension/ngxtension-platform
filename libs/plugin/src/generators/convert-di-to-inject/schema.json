{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert from constructor DI to inject function."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive/pipe you want to convert to use the inject function"
		},
		"includeReadonlyByDefault": {
			"type": "boolean",
			"aliases": ["includeReadonly", "includeReadonlyByDefault"],
			"description": "Whether to include readonly for injections by default. Default is false."
		},
		"useEsprivateFieldNotation": {
			"type": "boolean",
			"aliases": [
				"useESPrivateFieldNotation",
				"useEsprivateFieldNotation",
				"useEsPrivateFieldNotation"
			],
			"description": "Whether to replace TS private modifier with ES private field notation. Default is false."
		}
	}
}
