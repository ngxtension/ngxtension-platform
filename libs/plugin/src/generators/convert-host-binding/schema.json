{
	"$schema": "http://json-schema.org/schema",
	"$id": "ConvertHostBinding",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert from host binding to host property."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive you want to convert to use the host property."
		}
	}
}
