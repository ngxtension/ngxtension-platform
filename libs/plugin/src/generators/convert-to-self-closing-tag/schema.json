{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert to self closing tags."
		},
		"path": {
			"type": "string",
			"description": "Path to the component/directive you want to convert to self closing tags"
		}
	}
}
