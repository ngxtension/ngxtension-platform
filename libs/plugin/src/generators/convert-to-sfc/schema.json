{
	"$schema": "http://json-schema.org/schema",
	"$id": "",
	"title": "",
	"type": "object",
	"properties": {
		"project": {
			"type": "string",
			"aliases": ["name", "projectName"],
			"description": "Project for which to convert to single file components."
		},
		"path": {
			"type": "string",
			"description": "Path to the component to convert to a single file component."
		},
		"maxInlineTemplateLines": {
			"type": "number",
			"description": "If the template has more lines than this, it will be skipped.",
			"default": 200
		},
		"maxInlineStyleLines": {
			"type": "number",
			"description": "If the styles have more lines than this, they will be skipped.",
			"default": 200
		},
		"moveStyles": {
			"type": "boolean",
			"description": "Move styles to the single file component.",
			"default": false
		}
	}
}
