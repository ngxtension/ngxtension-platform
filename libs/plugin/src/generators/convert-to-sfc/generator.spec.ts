import { Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';

import convertToSFCGenerator from './generator';
import { ConvertToSFCGeneratorSchema } from './schema';

const template = `<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>`;

const styles = `h1 {
  color: red;
}`;

const filesMap = {
	notComponent: `
import { Injectable } from '@angular/core';

@Injectable()
export class MyService {}
`,
	componentNoTemplate: `
import { Component } from '@angular/core';

@Component({})
export class MyCmp {}
`,

	componentWithTemplateUrl: `
import { Component, Input } from '@angular/core';

@Component({
  templateUrl: './my-file.html'
})
export class MyCmp {
}
`,
	componentWithTemplateUrlAndStyleUrls: `
import { Component, Input } from '@angular/core';

@Component({
  templateUrl: './my-file.html',
  styleUrls: ['./my-file.css']
})
export class MyCmp {
}
`,
	componentWithTemplateUrlAndStyleUrl: `
import { Component, Input } from '@angular/core';

@Component({
  templateUrl: './my-file.html',
  styleUrl: './my-file.css'
})
export class MyCmp {
}
`,
	componentWithInlineTemplate: `
import { Component, Input } from '@angular/core';

@Component({
  template: \`
    <router-outlet></router-outlet>
  \`
})
export class MyCmp {
}
`,
} as const;

describe('convertToSFCGenerator', () => {
	let tree: Tree;
	const options: ConvertToSFCGeneratorSchema = {
		path: 'libs/my-file.ts',
		moveStyles: true,
		maxInlineTemplateLines: 10,
		maxInlineStyleLines: 10,
	};

	function setup(file: keyof typeof filesMap) {
		tree = createTreeWithEmptyWorkspace({ layout: 'apps-libs' });
		tree.write('package.json', `{"dependencies": {"@angular/core": "17.1.0"}}`);
		tree.write(`libs/my-file.ts`, filesMap[file]);
		tree.write(`libs/my-file.css`, styles);

		if (
			file === 'componentWithTemplateUrl' ||
			file === 'componentWithTemplateUrlAndStyleUrls' ||
			file === 'componentWithTemplateUrlAndStyleUrl'
		) {
			tree.write(`libs/my-file.html`, template);
			return () => {
				return [
					tree.read('libs/my-file.ts', 'utf8'),
					filesMap[file],
					tree.read('libs/my-file.html', 'utf8'),
					template,
					tree.read('libs/my-file.css', 'utf8'),
				];
			};
		}

		return () => {
			return [tree.read('libs/my-file.ts', 'utf8'), filesMap[file]];
		};
	}

	it('should not do anything if not component/directive', async () => {
		const readContent = setup('notComponent');
		await convertToSFCGenerator(tree, options);
		const [updated, original] = readContent();
		expect(updated).toEqual(original);
	});

	it('should not do anything if no template', async () => {
		const readContent = setup('componentNoTemplate');
		await convertToSFCGenerator(tree, options);
		const [updated, original] = readContent();
		expect(updated).toEqual(original);
	});

	it('should convert properly for templateUrl', async () => {
		const readContent = setup('componentWithTemplateUrl');
		await convertToSFCGenerator(tree, options);
		const [updated, , updatedHtml] = readContent();
		expect(updated).toMatchSnapshot();
		expect(updatedHtml).toMatchSnapshot();
	});

	it('should convert properly for templateUrl and styleUrls', async () => {
		const readContent = setup('componentWithTemplateUrlAndStyleUrls');
		await convertToSFCGenerator(tree, options);
		const [updated, , updatedHtml, updatedStyles] = readContent();
		expect(updated).toMatchSnapshot();
		expect(updatedHtml).toMatchSnapshot();
		expect(updatedStyles).toMatchSnapshot();
	});

	it('should convert properly for templateUrl and styleUrl', async () => {
		const readContent = setup('componentWithTemplateUrlAndStyleUrl');
		await convertToSFCGenerator(tree, options);
		const [updated, , updatedHtml, updatedStyles] = readContent();
		expect(updated).toMatchSnapshot();
		expect(updatedHtml).toMatchSnapshot();
		expect(updatedStyles).toMatchSnapshot();
	});

	it('should skip components with inline templates', async () => {
		const readContent = setup('componentWithInlineTemplate');
		await convertToSFCGenerator(tree, options);
		const [updated, , updatedHtml] = readContent();
		expect(updated).toMatchSnapshot();
		expect(updatedHtml).toMatchSnapshot();
	});
});
