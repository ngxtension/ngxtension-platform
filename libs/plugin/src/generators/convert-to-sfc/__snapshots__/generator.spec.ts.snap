// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertToSFCGenerator should convert properly for templateUrl 1`] = `
"
import { Component, Input } from '@angular/core';

@Component({
  template: \`
<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>
\`
})
export class MyCmp {
}
"
`;

exports[`convertToSFCGenerator should convert properly for templateUrl 2`] = `null`;

exports[`convertToSFCGenerator should convert properly for templateUrl and styleUrl 1`] = `
"
import { Component, Input } from '@angular/core';

@Component({
  template: \`
<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>
\`,
  styles: \`
h1 {
  color: red;
}
\`
})
export class MyCmp {
}
"
`;

exports[`convertToSFCGenerator should convert properly for templateUrl and styleUrl 2`] = `null`;

exports[`convertToSFCGenerator should convert properly for templateUrl and styleUrl 3`] = `
"<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>"
`;

exports[`convertToSFCGenerator should convert properly for templateUrl and styleUrls 1`] = `
"
import { Component, Input } from '@angular/core';

@Component({
  template: \`
<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>
\`,
  styles: \`
h1 {
  color: red;
}
\`
})
export class MyCmp {
}
"
`;

exports[`convertToSFCGenerator should convert properly for templateUrl and styleUrls 2`] = `null`;

exports[`convertToSFCGenerator should convert properly for templateUrl and styleUrls 3`] = `
"<div>Hello</div>
<app-my-cmp1>123</app-my-cmp1>"
`;

exports[`convertToSFCGenerator should skip components with inline templates 1`] = `
"
import { Component, Input } from '@angular/core';

@Component({
  template: \`
    <router-outlet></router-outlet>
  \`
})
export class MyCmp {
}
"
`;

exports[`convertToSFCGenerator should skip components with inline templates 2`] = `undefined`;
